### ISSUE TITLE
When Undotree window is open, executing macro with a count causes multiple changes

### ISSUE THREAD

**Issue by Asheq** [mbbill/undotree#111]
# Steps to reproduce
1. Open Undotree window with `:UndotreeShow`
1. Type `qq` to record macro in register q
1. Type `othis is a test<Esc>q` to type text in a new line and stop recording macro
1. Type `5@q` to execute macro 5 times.

**Actual**: Five new changes are added to the undo list
**Expected**: One change is added to the undo list (this is how it works when the Undotree window is closed)

Video:
![example](https://user-images.githubusercontent.com/4657140/72102280-6f9ce000-32db-11ea-928a-82e1d55502bf.gif)

**Comment by mbbill** [comment]
this is interesting. Looks like some of the auto commands might changed the undo behaviour.
btw, thanks for the video! it's super easy to repro the issue on my side thanks to it.

**Comment by MXfive** [comment]
Was this ever resolved?  I am seeing this is on nvim 0.10 nightly with latest undo tree.

**Comment by mbbill** [comment]
> Was this ever resolved? I am seeing this is on nvim 0.10 nightly with latest undo tree.

it's not

**Comment by MXfive** [comment]
Any idea what mere the issue might be?  Interesting in having a stab at this at some point.  Currently having to restart without Undotree loaded when I want to run macros like this.

**Comment by mbbill** [comment]
initially I think it might due to some autocommands that triggered a undo node creation, like exiting/entering insertion mode will do that. But that's only when the undotree panel is open. If you have to disable the plugin then I believe something else is wrong because when undotree panel is not open the auto commands shouldn't be registered. Do you mind check the auto commands by running `:au BufEnter` or simply `:au` and check if there's anything related to undotree?

**Comment by MXfive** [comment]
Nothing that could be related to undotree on `BufEnter` for my setup:


```
--- Autocommands ---
FileExplorer  BufEnter
*sil call s:LocalBrowse(expand("<amatch>"))
lualine  BufEnter
*lua require'lualine.components.branch.git_branch'.find_git_dir()
lua require'lualine.components.diff.git_diff'.update_diff_args()
lualine_stl_refresh  BufEnter
*call v:lua.require'lualine'.refresh({'kind': 'tabpage', 'place': ['statusline'], 'trigger': 'autocmd'})
treesitter_context_update  BufEnter
*<Lua 204: ~/.local/share/nvim/lazy/nvim-treesitter-context/lua/treesitter-context.lua:21>
```

**Comment by mbbill** [comment]
undotree gets opened? is there any hotkey that opens the panel getting accidently recorded?

**Comment by MXfive** [comment]
> undotree gets opened? is there any hotkey that opens the panel getting accidently recorded?

Nope, was able to reproduce reliably.  Was just doing a find and replace of an array attribute across multiple files.  Undotree did not open while recording the macro, nor while manually replaying it.  But it did when using `:argdo ...`

**Comment by mbbill** [comment]
that's interesting...

**Comment by MXfive** [comment]
Not sure what happened, but now I am really unable to reproduce it..
